### PR TITLE
expose doc store cache size

### PR DESCRIPTION
**Pull request by PSeitz**
expose lru doc store cache size
optimize doc store cache size

### CODE REVIEW

File: src/core/searcher.rs [hunk 1]
```diff
@@ -77,11 +77,17 @@ impl Searcher {
...
         generation: TrackedObject<SearcherGeneration>,
+        doc_store_cache_size: usize,
     ) -> io::Result<Searcher> {
         let store_readers: Vec<StoreReader> = segment_readers
             .iter()
             .map(SegmentReader::get_store_reader)
             .collect::<io::Result<Vec<_>>>()?;
+
+        for store_reader in &store_readers {
+            store_reader.set_cache_size(doc_store_cache_size);
```
Comment (fulmicoton) [inline src/core/searcher.rs]: why could not this be done at the SegmentReader creation?

File: src/store/reader.rs [hunk 2]
```diff
@@ -112,6 +144,11 @@ impl StoreReader {
         self.cache.stats()
     }
 
+    /// Set lru cache size for decompressed blocks. Defaults to 100 (LRU_CACHE_CAPACITY).
+    pub(crate) fn set_cache_size(&self, size: usize) {
+        self.cache.set_size(size)
```
Comment (fulmicoton) [inline src/store/reader.rs]: why is this a setter?

**Comment by fulmicoton** [comment]
that PR is allowing to set the cache size on the docstore. The PR does not really explain why.
My understanding is that the benefit is to allow a cache of size 1 when merging?

Could this be done on segment reader creation? 
Or not done at all actually? Are you thinking of enabling a large LRU size?

File: src/core/searcher.rs (hunk 1 above)
Comment (PSeitz) [reply src/core/searcher.rs]: yes I changed it

**Comment by PSeitz** [comment]
> that PR is allowing to set the cache size on the docstore. The PR does not really explain why. My understanding is that the benefit is to allow a cache of size 1 when merging?
> 
> Could this be done on segment reader creation? Or not done at all actually? Are you thinking of enabling a large LRU size?

One benefit is to reduce the cache size for merging. The other benefit is to reduce memory consumption for quickwit fetch_docs. The cache size is much more important with the larger blocks now.

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**